src/main/java/org/sunbird/migrate/service/impl/UserMigrationServiceImpl.java

```diff
@@ -0,0 +1,272 @@
+import org.sunbird.common.util.PropertiesCache;
+import org.sunbird.core.config.PropertiesConfig;
+import org.sunbird.migrate.service.UserMigrationService;
+import org.sunbird.profile.service.ProfileService;
+
+import java.text.SimpleDateFormat;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+
+@Service
+public class UserMigrationServiceImpl implements UserMigrationService {
+
+    private Logger log = LoggerFactory.getLogger(getClass().getName());
+
+    @Autowired
+    OutboundRequestHandlerServiceImpl outboundRequestHandlerService;
+
+    @Autowired
+    CbExtServerProperties serverConfig;
+
+    @Autowired
+    PropertiesConfig propertiesConfig;
+
+    @Autowired
+    ProfileService profileService;
+
+    @Override
+    public SBApiResponse migrateUsers(String authToken) {
+        SBApiResponse response = new SBApiResponse(Constants.API_USER_MIGRATION);
+        StringBuilder url = new StringBuilder(propertiesConfig.getLmsServiceHost()).append(propertiesConfig.getLmsUserSearchEndPoint());
+        log.info("Printing user search URL: {}", url);
+
+        int offset = 0;
+        int limit = 250;
+        boolean allOperationsSuccessful = true;
+        String custodianOrgName = serverConfig.getCustodianOrgName();
+        String custodianOrgId   = serverConfig.getCustodianOrgId();
+
+        try {
+            while (true) {
+                Map<String, Object> request = userSearchRequestBody(offset, limit);
+                Map<String,String> headerMap = createHeaderMap(authToken);
+                Map<String, Object> updateResponse = outboundRequestHandlerService.fetchResultUsingPost(url.toString(), request, headerMap);
+
+                if (Constants.OK.equalsIgnoreCase((String) updateResponse.get(Constants.RESPONSE_CODE))) {
+                    Map<String, Object> result = (Map<String, Object>) updateResponse.get(Constants.RESULT);
+                    Map<String, Object> responseData = (Map<String, Object>) result.get(Constants.RESPONSE);
+                    List<Map<String, Object>> users = (List<Map<String, Object>>) responseData.get(Constants.CONTENT);
+
+                    if (users.isEmpty()) {
+                        log.info("No more users found. Exiting pagination.");
+                        break;
+                    }
+
+                    for (Map<String, Object> user : users) {
+                        List<Map<String, Object>> organisations = (List<Map<String, Object>>) user.get(Constants.ORGANISATIONS);
+                        String userId = (String) user.get(Constants.USER_ID);
+                        boolean orgFound = false;
+
+                        if (organisations != null) {
+                            orgFound = organisations.stream()
+                                    .anyMatch(organisation -> custodianOrgName.equalsIgnoreCase((String) organisation.get(Constants.ORG_NAME)));
```

Comment:
No... we shouldn't check the organisations array list... 

directly check the user.rootOrgName -- this is the value which gives the current user's organisation...